modules/nf-core/yahs/main.nf

```diff
@@ -0,0 +1,44 @@
+process YAHS {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::yahs=1.2a.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/yahs:1.2a.2--h7132678_0':
+        'quay.io/biocontainers/yahs:1.2a.2--h7132678_0' }"
+
+    input:
+    tuple val(meta), path(bed)
+    tuple path(ref), path(fai)
+    val(if_break)
+    val(motif)
+    val(resolutions)
+
+    output:
+    tuple val(meta), path("yahs.out_scaffolds_final.fa"), emit: scaffolds_fasta
+    tuple val(meta), path("yahs.out_scaffolds_final.agp"), emit: scaffolds_agp
+    tuple val(meta), path("yahs.out.bin"), emit: binary
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def break_param = if_break ? '' : ' --no-contig-ec'
```

Comment:
As referenced above, I think all the options here can come directly through from ext.args without the need for explicit `val` inputs.